.github/workflows/ci.yml

```diff
@@ -137,8 +150,21 @@ jobs:
 
     - run: nox --python ${{ matrix.python.action }} -e ${{ matrix.task.nox }} -- --use-wheel dist/*.whl
 
+    - name: Store coverage file
+      uses: actions/upload-artifact@v3
+      with:
+        # We have a single artifact shared for each workflow run.
+        # in this way we should intefere with coverage reports from other PRs
+        # or previous runs for the same PR.
+        name: coverage-${{ github.run_id }}
+        # Is important to keep the unique names for the coverage files
+        # so that when uploaded to the same artifact, they don't overlap.
```

Comment:
```suggestion
        # It is important to keep the unique names for the coverage files
        # so that when uploaded to the same artifact, they don't overlap.
```